db/view/view.cc

```diff
@@ -1755,6 +1755,39 @@ future<> view_update_generator::mutate_MV(
     });
 }
 
+
+utils::small_vector<gms::inet_address, 2> view_update_generator::get_endpoints_for_view_update(
+        dht::token base_token,
+        const frozen_mutation_and_schema& view_update) const {
+    dht::token view_token = dht::get_token(*view_update.s, view_update.fm.key());
+    const sstring& keyspace_name = view_update.s->ks_name();
+
+    // For every base table replica there's a single paired view replica.
+    std::optional<gms::inet_address> paired_replica_endpoint =
+        get_view_natural_endpoint(_proxy.local().local_db(), keyspace_name, base_token, view_token);
+
+    // In case of an ongoing topology change it might also be necessary to send the update to pending nodes.
+    inet_address_vector_topology_change pending_endpoints = _proxy.local()
+                                                                .local_db()
+                                                                .find_keyspace(keyspace_name)
+                                                                .get_effective_replication_map()
```

Comment:
It's wrong to use get_effective_replication_map() for a point query. Its results are only valid for as long as the pointer is held.

Correct use it:
 1. at the beginning of the operation, capture the effective replication map
 2. use it for all such queries
 3. at the end of the operation (including acknowledgement from remotes), release the erm_ptr

Reply:
This allows the rest of the system to avoid changing the topology to an incompatible version while the operation is in progress.

Reply:
Yes, this is a bit "incorrect", but it is a conscious decision.
This function is used only to calculate the throttling delay, which doesn't need to be perfect, it's a heuristic.
There's no need to block the whole schema change for this.

I agree that `mutate_MV` should use the `effective_replication_map` (it doesn't at the moment), but there's more risks there, as each view update has a 5 minute timeout, so a schema change could be blocked for the whole 5 minutes.